codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/XmlShapeDeserVisitor.java

```diff
@@ -327,6 +326,8 @@ protected void deserializeStructure(GenerationContext context, StructureShape sh
             writer.write("t, done, err := decoder.Token()");
             writer.write("if err != nil { return err }");
             writer.write("if done { break }");
+            // TODO: fix self closing tags deser
+            writer.write("_ = t");
```

Comment:
Is this still outstanding?

Reply:
yeah for now we don't know the behavior in case of self closing tags for numbers, boolean, time etc. 

That will be a separate follow up PR that touches XML as well and shouldn't block this PR.